src/iarc7_motion/iarc_tasks/track_roomba_task.py

```diff
@@ -147,11 +154,24 @@ def get_desired_command(self):
                 y_overshoot = overshoot * math.sin(math.atan2(roomba_y_velocity, roomba_x_velocity )
                                                         + math.atan2(self._y_overshoot, self._x_overshoot))
 
-                # p-controller
-                x_vel_target = ((self._roomba_point.point.x + x_overshoot)
-                                    * self._K_X + roomba_x_velocity)
-                y_vel_target = ((self._roomba_point.point.y + y_overshoot)
-                                    * self._K_Y + roomba_y_velocity)
+                x_diff = self._roomba_point.point.x + x_overshoot
+                y_diff = self._roomba_point.point.y + y_overshoot
+
+                x_success, x_vel_target = self._x_pid.update(x_diff, rospy.Time.now(), False)
+                y_success, y_vel_target = self._y_pid.update(y_diff, rospy.Time.now(), False)
+
+                # PID controller does setpoint - current; 
+                # the difference from do_transform_point is from the drone to the roomba, 
+                # which is the equivalent of current-setpoint, so take the negative response
+                if x_success: 
+                    x_vel_target = -x_vel_target + roomba_x_velocity
```

Comment:
You should use different names for the `x_vel_target` that you set earlier and this one here

Reply:
fixed